### ISSUE TITLE
Drop the ServerHandler class

### ISSUE THREAD

**Issue by natebosch** [dart-lang/shelf#205]
It seems to exist to allow a late binding between a `Server` and a `Handler`

**Comment by kevmoo** [comment]
Agreed. I asked about this a while ago and came to a similar conclusion!
